neuralogic/inference/evaluation_inference_engine.py

```diff
@@ -30,7 +30,7 @@ def query(self, query: BaseRelation, examples: Optional[List[Union[BaseRelation,
 
         try:
             built_dataset = self.model.build_dataset(self.dataset)
-            results = self.model(built_dataset.samples, train=False)
+            results = self.model.test(built_dataset.samples)
```

Comment:
Remove evaluation inference engine, as it is not relevant anymore -> replaced by sample

Reply:
Maybe try to add more efficient substitution look up for the query?